### PR TITLE
Change delete plan

### PR DESCRIPTION
**Pull request by ouyuanning**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3931

## What this PR does / why we need it:
1、 only rowid will return in delete plan now.
2、change Delete interface Of Relation. we need more columns for input in future(when we get partition columns). so change arg2 from Vector to Batch

### CODE REVIEW

**Review by nnsgmsone — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

File: pkg/vm/engine/disttae/table.go [hunk 1]
```diff
@@ -191,9 +190,9 @@ func (tbl *table) Update(ctx context.Context, bat *batch.Batch) error {
 	return nil
 }
 
-func (tbl *table) Delete(ctx context.Context, vec *vector.Vector, name string) error {
-	bat := batch.NewWithSize(1)
-	bat.Vecs[0] = vec
+func (tbl *table) Delete(ctx context.Context, bat *batch.Batch, name string) error {
+	// bat := batch.NewWithSize(1)
+	// bat.Vecs[0] = vec
 	bats, err := partitionBatch(bat, tbl.insertExpr, tbl.db.txn.proc, len(tbl.parts))
```
Comment (dongdongyang33) [inline pkg/vm/engine/disttae/table.go:196]: maybe should use `tbl.deleteExpr` here?

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/table.go:196]: yes

**Comment by nnsgmsone** [comment]
The following sql, the number of columns written is still equal to 1.
```
create database test;
use test;
create table t (a int);

begin;
insert into t values(1);
select * from t;
delete from t where a = 1;

**Review by reusee — APPROVED** [review]

**Review by fengttt — COMMENTED** [review]
Why does delete need more columns?   Isn't the Rowid column enough?

**Comment by nnsgmsone** [comment]
> Why does delete need more columns? Isn't the Rowid column enough?

Need to determine which dn to pass this delete request to. . rowid does not contain parition information, so additional information is needed (e.g. primary key to determine which dn to send the request to)

**Comment by ouyuanning** [comment]
> delete from t where a = 1;

that's a bug. fixed

**Review by aunjgr — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/colexec/auto_incr.go [hunk 2]
```diff
@@ -286,7 +286,7 @@ func getCurrentIndex(param *AutoIncrParam, colName string) (uint64, uint64, erro
 
 func updateAutoIncrTable(param *AutoIncrParam, curNum uint64, name string, mp *mpool.MPool) error {
 	bat := makeAutoIncrBatch(name, curNum, 1, mp)
-	err := param.rel.Delete(param.ctx, bat.GetVector(0), AUTO_INCR_TABLE_COLNAME[0])
```
Comment (nnsgmsone) [inline pkg/sql/colexec/auto_incr.go:289]: Here 3 columns are passed in and none of the columns are rowid.

Comment (ouyuanning) [reply pkg/sql/colexec/auto_incr.go:289]: get it.  I will fix it.

Comment (ouyuanning) [reply pkg/sql/colexec/auto_incr.go:289]: @jianwan0214 had fixed it. thanks @jianwan0214

**Review by dongdongyang33 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**